src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -71,12 +71,15 @@ public DataFetcherResult getRecords(int maxRecords) {
         
         if (nextIterator != null) {
             try {
-                return new AdvancingResult(kinesisProxy.get(nextIterator, maxRecords));
+                GetRecordsResult getRecordsResult = kinesisProxy.get(nextIterator, maxRecords);
+                LOG.info("Successfully fetched records from Kinesis for shard " + shardId + ": " + getRecordsResult);
+                return new AdvancingResult(getRecordsResult);
             } catch (ResourceNotFoundException e) {
                 LOG.info("Caught ResourceNotFoundException when fetching records for shard " + shardId);
                 return TERMINAL_RESULT;
             }
         } else {
+            LOG.info("Cannot fetch records from Kinesis for shard " + shardId + ": nextIterator is null.");
```

Comment:
This is more of a skipping to fetch case, instead of can not. Re-phrasing would be nice.

Reply:
Good callout - rephrased the log